crates/bevy_input/src/gamepad.rs

```diff
@@ -251,14 +304,14 @@ pub fn gamepad_connection_system(
     mut gamepad_event: EventReader<GamepadEvent>,
 ) {
     for event in gamepad_event.iter() {
-        match &event {
-            GamepadEvent(gamepad, GamepadEventType::Connected) => {
-                gamepads.register(*gamepad);
-                info!("{:?} Connected", gamepad);
+        match event.event_type {
```

Comment:
This is a very nice change, I love it!

Reply:
Yes same. It makes the code a lot cleaner and clearer.